### DISCUSSION TITLE
Bun support / fixes

### DISCUSSION THREAD

**Discussion by joas8211** [panva/node-oidc-provider#1274] [Ideas]
I want to use this library with Bun runtime.

There's already mention of Bun in [the runtime compability check](https://github.com/panva/node-oidc-provider/blob/main/lib/index.js#L7) but it does not really do anything with Bun v1.1.24 because value of `process.release.lts` is `false`.  I don't know what it was meant to do, but if Bun is supported then it could be maybe fixed.

Then there's a little difference how functions are stringified which causes TTL configuration validation to fail. But it is easy to fix.

<details><summary>The Fix</summary>

```diff
checkTTL() {
  Object.entries(this.ttl).forEach(([key, value]) => {
    let valid = false;
    switch (typeof value) {
      case 'function':
-       if (value.constructor.toString() === 'function Function() { [native code] }') {
+       if (value.constructor.name === "Function") {
          valid = true;
        }
        break;
      case 'number':
        if (Number.isSafeInteger(value) && value > 0) {
          valid = true;
        }
        break;
      case 'undefined': // does not expire
        valid = true;
        break;
      default:
    }

    if (!valid) {
      throw new TypeError(`ttl.${key} must be a positive integer or a regular function returning one`);
    }
  });
}
```
</details> 

I didn't immidiately find anything else that was broken with Bun. So could the fix be at least applied, and maybe even make Bun officially supported?

**Comment by panva** [comment]
https://github.com/panva/node-oidc-provider/pull/1253#issuecomment-2006097445

#1250 

I will not be supporting any runtime other than Node.js in this module.

**Reply by panva** [reply]
More importantly

> I want to use this library with Bun runtime.

Why? Bun's `node:crypto` module is still in its infancy and full of slow questionable polyfills. Is that what you want to back your OAuth server deployment with?

**Reply by joas8211** [reply]
In fact, your comment ignited the idea to convert the project to Node.js which might be clever anyway. I was hoping, since the project is in it's early phase that Bun gets mature enough in the time of the development. But maybe it doesn't make sense to rely on interoperability if it's possible to use the actual runtime to run application reliably.

**Reply by panva** [reply]
💯
